syntax = "proto3";
package proto;

message response {
  uint32 id = 1;                   
  string special_code = 2;         
  uint32 timestamp1 = 3;             
  uint32 value_a = 4;               
  uint32 status_code = 5;           
  uint32 sub_type = 6;               
  uint32 version = 7;              
  uint32 level = 8;                 
  uint32 flags = 9;                
  string welcome_message = 12;     
  string region = 13;                
  string json_metadata = 14;         
  string big_numbers = 15;          
  uint32 balance = 20;              
  uint32 score = 22;                
  uint32 upgrades = 33;             
  uint32 achievements = 35;         
  uint32 total_playtime = 36;        
  uint32 energy = 37;               
  uint32 rank = 38;                 
  uint32 xp = 39;                   
  uint32 timestamp2 = 40;            
  uint32 error_code = 41;           
  uint32 last_active = 44;           
  
  GuildInfo guild_details = 47;      
  
  string empty_field = 49;          

  message GuildInfo {
    string region = 1;           
    uint32 clan_id = 2;        
    uint32 members_online = 3;     
    uint32 total_members = 4;       
    uint32 regional  = 5;        
    uint32 reward_time = 6;         
    uint32 expire_time = 7;      
  }
}